src/test/java/org/broadinstitute/hellbender/utils/config/ConfigIntegrationTest.java

```diff
@@ -67,54 +109,94 @@ public String getTestedClassName() {
     @Test(dataProvider = "provideForTestToolWithConfigValidation")
     public void testToolWithConfigValidation(final Path configFilePath) {
 
-        final File tmpFile = getSafeNonExistentFile("ConfigIntegrationTest.output.txt");
-        tmpFile.deleteOnExit();
-
-        // Create some arguments for our command:
-        final ArgumentsBuilder args = new ArgumentsBuilder();
-        args.add("--" + StandardArgumentDefinitions.GATK_CONFIG_FILE_OPTION);
-        args.add(configFilePath.toUri().toString());
-        args.add("-" + StandardArgumentDefinitions.OUTPUT_SHORT_NAME);
-        args.add(tmpFile);
+        // Grab the properties so we can reset them later:
+        // NOTE: CLONE HERE IS ABSOLUTELY NECESSARY OR THIS WILL FAIL VERY VERY HARD!
+        final Properties systemProperties = (Properties)System.getProperties().clone();
+//        System.out.println("================================================================================");
+//        listProperties(systemProperties);
+//        System.out.println("================================================================================");
+
+        try {
+            final File tmpConfigPropsFile = getSafeNonExistentFile("ConfigIntegrationTest.config.properties");
+            final File tmpSystemPropsFile = getSafeNonExistentFile("ConfigIntegrationTest.system.properties");
+
+            // Add in some system options.
+            // None of these should be masked by the options of the same name in the config file.
+            final Map<String, String> systemPropertiesToQuery = new HashMap<>();
+            systemPropertiesToQuery.put("gatk_stacktrace_on_user_exception", "true");
+            systemPropertiesToQuery.put("samjdk.compression_level", "7777");
+
+            for (final Map.Entry<String, String> entry : systemPropertiesToQuery.entrySet()) {
+                System.setProperty(entry.getKey(), entry.getValue());
+            }
+
+            // Create some arguments for our command:
+            final ArgumentsBuilder args = new ArgumentsBuilder();
+            args.add("--" + StandardArgumentDefinitions.GATK_CONFIG_FILE_OPTION);
```

Comment:
these could be a little bit less verbose if you use `args.addArgument(key, value)`
and there's a special `args.addOutput(file)` too

it's fine if you prefer them this way though

Reply:
Sounds good.  I didn't realize that was an option.